dot/types/authority.go

```diff
@@ -128,6 +129,24 @@ type AuthorityAsAddress struct {
 	Weight  uint64
 }
 
+// UnmarshalJSON converts data to Go struct of type AuthorityAsAddress.
+func (a *AuthorityAsAddress) UnmarshalJSON(buf []byte) error {
+	tmp := []interface{}{&a.Address, &a.Weight}
```

Comment:
maybe add a comment that it's encoded as an array `[]` instead of an object `{}`, which is why this need.